src/algebra/lattice_ordered_group.lean

```diff
@@ -0,0 +1,585 @@
+        { apply and.right
+            (iff.elim_left (le_iff_pos_le_neg_ge _ _)
+            (lattice_ordered_comm_group.le_abs a)), } },
+      { apply lattice_ordered_comm_group.neg_pos, }
+    end,
+    rw [← inf_mul_sup, pos_inf_neg_eq_one, one_mul, ← mod_eq_pos ],
+    apply sup_le,
+    apply and.left
+      (iff.elim_left (le_iff_pos_le_neg_ge _ _)
+      (lattice_ordered_comm_group.le_abs a)),
+    rw neg_eq_pos_inv,
+    apply and.left
+      (iff.elim_left (le_iff_pos_le_neg_ge _ _)
+      (lattice_ordered_comm_group.inv_le_abs a)), }
+end
+
+/--
+Let `α` be a lattice ordered commutative group, let `a` and `b` be elements in `α` and let `|b - a|`
+be the absolute value of `b - a`. Then,
+$$a ⊔ b - (a ⊓ b) = |b - a|.$$
+-/
+@[to_additive]
+lemma sup_div_inf_eq_abs_div (a b : α) : (a ⊔ b) / (a ⊓ b) = |b / a| :=
+begin
+  rw [sup_eq_mul_pos_div, inf_comm, inf_eq_div_pos_div, div_eq_mul_inv],
+  nth_rewrite 1 div_eq_mul_inv,
+  rw [mul_inv_rev, inv_inv, mul_comm, ← mul_assoc, inv_mul_cancel_right, pos_eq_neg_inv (a / b)],
+  nth_rewrite 1 div_eq_mul_inv,
+  rw [mul_inv_rev, ← div_eq_mul_inv, inv_inv, ← pos_mul_neg],
+end
+
+/--
+Let `α` be a lattice ordered commutative group, let `a` and `b` be elements in `α` and let `|b - a|`
+be the absolute value of `b - a`. Then,
+$$2•(a ⊔ b) = a + b + |b - a|.$$
+-/
+@[to_additive]
+lemma sup_sq_eq_mul_mul_abs_div (a b : α) : (a ⊔ b)^2 = a * b * |b / a| :=
+begin
+  rw [← inf_mul_sup a b, ← sup_div_inf_eq_abs_div, div_eq_mul_inv, ← mul_assoc, mul_comm,
+    mul_assoc, ← pow_two, inv_mul_cancel_left],
+end
+
+/--
+Let `α` be a lattice ordered commutative group, let `a` and `b` be elements in `α` and let `|b-a|`
+be the absolute value of `b-a`. Then,
+$$2•(a ⊓ b) = a + b - |b - a|.$$
+-/
+@[to_additive]
+lemma two_inf_eq_mul_div_abs_div (a b : α) : (a ⊓ b)^2 = a * b / |b / a| :=
+begin
+  rw [← inf_mul_sup a b, ← sup_div_inf_eq_abs_div, div_eq_mul_inv, div_eq_mul_inv,
+    mul_inv_rev, inv_inv, mul_assoc, mul_inv_cancel_comm_assoc, ← pow_two],
+end
+
+/--
+Every lattice ordered commutative group is a distributive lattice
+-/
+@[to_additive, priority 100] -- see Note [lower instance priority]
+instance lattice_ordered_comm_group_to_distrib_lattice (α : Type u)
+  [s: lattice α] [comm_group α] [covariant_class α α (*) (≤)] : distrib_lattice α :=
+{ le_sup_inf :=
+  begin
+    intros,
+    rw [← mul_le_mul_iff_left (x ⊓ (y ⊓ z)),  inf_mul_sup x (y ⊓ z),
+      ← inv_mul_le_iff_le_mul, le_inf_iff ],
+    split,
+    { rw [inv_mul_le_iff_le_mul, ← inf_mul_sup x y ],
+      apply mul_le_mul',
+      { apply inf_le_inf_left, apply inf_le_left, },
+      { apply inf_le_left, } },
+    { rw [inv_mul_le_iff_le_mul, ← inf_mul_sup x z ],
+      apply mul_le_mul',
+      { apply inf_le_inf_left, apply inf_le_right, },
+      { apply inf_le_right, }, }
+  end,
+  ..s
+}
+
+@[to_additive]
+lemma div_mul_div_eq_mul_div_mul (x y w z : α) : x / y * (w / z) = x * w / (y * z) :=
```

Comment:
This lemma isn't specific to this file, right? Please move it to its natural home.

Reply:
I'd wondered about that... there seem to be more existing lemmas involving `sub` in than there are `div` in mathlib. Should the `sub` results be replaced by `div` equivalents and `to_additive`?

Reply:
I think we already have this lemma for subtraction somewhere else. We certainly have it for addition as `add_add_add_comm`.

Reply:
The closest analogue I could find was [add_sub_comm](https://github.com/leanprover-community/mathlib/blob/master/src/algebra/group/basic.lean#L488) so I've added it to that file as `div_mul_comm`. I did look into converting the other statements in `algebra/group/basic` into multiplicative in a separate PR, but there are then name collisions with `algebra/group_with_zero`.